easybuild/easyblocks/s/siesta.py

```diff
@@ -0,0 +1,342 @@
+                # Needed for a couple of the utils
+                (r"^(COMP_LIBS\s*=.*)$", r"\1\nWXML = libwxml.a"),
+                (r"^(FFLAGS\s*=\s*).*$", r"\1 -fPIC %s" % os.environ['FCFLAGS']),
+            ])
+
+            if netcdff_loc:
+                regex_subs.extend([
+                    (r"^(COMP_LIBS\s*=.*)$", r"\1\nNETCDF_LIBS = -lnetcdff"),
+                    (r"^(LIBS\s*=.*)$", r"\1 $(NETCDF_LIBS)"),
+                    (r"^(FPPFLAGS\s*=.*)$", r"\1 -DCDF"),
+                ])
+
+        apply_regex_substitutions(arch_make, regex_subs)
+
+        run_cmd('make %s' % par, log_all=True, simple=True, log_output=True)
+
+        # Put binary in temporary install dir
+        shutil.copy(os.path.join(self.cfg['start_dir'], 'Obj', 'siesta'),
+                    bindir)
+
+        if self.cfg['with_utils']:
+            # Make the utils
+            try:
+                os.chdir(os.path.join(start_dir, 'Util'))
+            except OSError, err:
+                raise EasyBuildError("Failed to change to Util dir: %s", err)
+
+            # clean_all.sh might be missing executable bit...
+            adjust_permissions('./clean_all.sh', stat.S_IXUSR, recursive=False, relative=True)
+            run_cmd('./clean_all.sh', log_all=True, simple=True, log_output=True)
+
+            if LooseVersion(self.version) >= LooseVersion("4.1"):
+                regex_subs_TS = [
+                    (r"^default:.*$", r""),
+                    (r"^EXE\s*=.*$", r""),
+                    (r"^(include\s*..ARCH_MAKE.*)$", r"EXE=tshs2tshs\ndefault: $(EXE)\n\1"),
+                    (r"^(INCFLAGS.*)$", r"\1 -I%s" % obj_dir),
+                ]
+
+                apply_regex_substitutions(os.path.join(self.cfg['start_dir'], 'Util', 'TS', 'tshs2tshs', 'Makefile'), regex_subs_TS)
+
+            # SUFFIX rules in wrong place
+            regex_subs_suffix = [
+                (r'^(\.SUFFIXES:.*)$', r''),
+                (r'^(include\s*\$\(ARCH_MAKE\).*)$', r'\1\n.SUFFIXES:\n.SUFFIXES: .c .f .F .o .a .f90 .F90'),
+            ]
+            apply_regex_substitutions(os.path.join(self.cfg['start_dir'], 'Util', 'Sockets', 'Makefile'), regex_subs_suffix)
+            apply_regex_substitutions(os.path.join(self.cfg['start_dir'], 'Util', 'SiestaSubroutine', 'SimpleTest', 'Src', 'Makefile'), regex_subs_suffix)
+
+            regex_subs_UtilLDFLAGS = [
+                (r'(\$\(FC\)\s*-o\s)', r'$(FC) %s %s -o ' % (os.environ['FCFLAGS'], os.environ['LDFLAGS'])),
+            ]
+            apply_regex_substitutions(os.path.join(self.cfg['start_dir'], 'Util', 'Optimizer', 'Makefile'), regex_subs_UtilLDFLAGS)
+            apply_regex_substitutions(os.path.join(self.cfg['start_dir'], 'Util', 'JobList', 'Src', 'Makefile'), regex_subs_UtilLDFLAGS)
+
+            run_cmd('./build_all.sh', log_all=True, simple=True, log_output=True)
+
+            # Now move all the built utils to the temp installdir
+            expected_utils = [
```

Comment:
ah, wait, this isn't the sanity check...

well, it'll fail anyway when we try to copy files that aren't there